pkgs/env-builder/default.nix

```diff
@@ -0,0 +1,79 @@
+# ============================================================================ #
+#
+#
+#
+# ---------------------------------------------------------------------------- #
+{
+  self,
+  stdenv,
+  sqlite,
+  pkg-config,
+  nlohmann_json,
+  nix,
+  boost,
+  argparse,
+  semver,
+  flox-pkgdb,
+  sqlite3pp,
+  # sql-builder,
+}: let
+  src = builtins.path {
+    path = "${self}/env-builder";
+    filter = name: type: let
+      bname = baseNameOf name;
+      # Tests require internet so there's no point in including them
+      ignores = ["out" "bin" "lib" "tests"];
+      ext = let
+        m = builtins.match ".*\\.([^.]+)" name;
+      in
+        if m == null
+        then ""
+        else builtins.head m;
+      ignoredExts = ["o" "so" "dylib" "nix"];
+      notIgnored =
+        (! (builtins.elem bname ignores))
+        && (! (builtins.elem ext ignoredExts));
+      notResult = (builtins.match "result(-*)?" bname) == null;
+    in
+      notIgnored && notResult;
+  };
+in
+  stdenv.mkDerivation {
+    pname = "flox-cpp";
+    version = builtins.replaceStrings ["\n"] [""] (builtins.readFile "${src}/version");
+
+    propagatedBuildInputs = [semver nix.dev boost];
+    nativeBuildInputs = [pkg-config];
+    buildInputs = [
+      sqlite.dev
+      nlohmann_json
+      argparse
+      flox-pkgdb
+      sqlite3pp
+      # sql-builder
+    ];
+    nix_INCDIR = nix.dev.outPath + "/include";
+    boost_CFLAGS = "-I" + boost.outPath + "/include";
+    pkgdb_CLFAGS = "-I" + flox-pkgdb.outPath + "/include";
```

Comment:
We have some other requieed flags, but this works for now.
If you run into issues with include paths let me know.

I'm on my phone so I can't write a patch now.

Reply:
so far everything builds, I'll report if i run into issues